src/Runner.Worker/ActionManager.cs

```diff
@@ -504,27 +502,83 @@ private async Task DownloadRepositoryActionAsync(IExecutionContext executionCont
                 executionContext.Output($"Download action repository '{repositoryReference.Name}@{repositoryReference.Ref}'");
             }
 
+            var configurationStore = HostContext.GetService<IConfigurationStore>();
+            var isHostedServer = configurationStore.GetSettings().IsHostedServer;
+            if (isHostedServer)
+            {
+                string apiUrl = GetApiUrl(executionContext, forceDotCom: true);
+                string archiveLink = BuildLinkToActionArchive(apiUrl, repositoryReference.Name, repositoryReference.Ref);
+                Trace.Info($"Download archive '{archiveLink}' to '{destDirectory}'.");
+                await DownloadRepositoryActionAsync(executionContext, archiveLink, destDirectory);
+            }
+            else
+            {
+                string apiUrl = GetApiUrl(executionContext);
+
+                // URLs to try:
+                var archiveLinks = new List<string> {
+                    // API_URL/repos/{repository}/zipball/{ref}  - A built-in action or an action the user has created, on their GHES instance
+                    BuildLinkToActionArchive(apiUrl, repositoryReference.Name, repositoryReference.Ref),
+                    // API_URL/repos/actions/community-{repository.replace(/,_)}/zipball/{ref}  - A community action, synced to their GHES instance
+                    BuildLinkToActionArchive(apiUrl, $"actions/community-{repositoryReference.Name.Replace("/", "_")}", repositoryReference.Ref)
```

Comment:
For example: https://my-ghes/api/v3/repos/community-actions/some-org_some-action/tarball/v1